src/Strings.f90

```diff
@@ -1030,6 +1036,38 @@ PURE SUBROUTINE assign_StringType2A_to_StringType2A(thisStr,s)
     ENDSUBROUTINE assign_StringType2A_to_StringType2A
 !
 !-------------------------------------------------------------------------------
+!> @brief Assigns a short integer to a StringType.
+!> @param thisStr the string object
+!> @param i short integer
+!>
+!> The intent is that this will overload the assignment operator so a short
+!> (32-bit) integer can be assigned to a StringType.
+!>
+    PURE SUBROUTINE assign_Short_Integer_to_StringType(thisStr,i)
+      TYPE(StringType),INTENT(INOUT) :: thisStr
+      INTEGER(SNK),INTENT(IN) :: i
+      CHARACTER(LEN=11) :: l
+      WRITE(l,'(i11)') i
+      thisStr=TRIM(ADJUSTL(l))
+    ENDSUBROUTINE assign_Short_Integer_to_StringType
+!
+!-------------------------------------------------------------------------------
+!> @brief Assigns a long integer to a StringType.
+!> @param thisStr the string object
+!> @param i long integer
+!>
+!> The intent is that this will overload the assignment operator so a long
+!> (64-bit) integer can be assigned to a StringType.
+!>
+    PURE SUBROUTINE assign_Long_Integer_to_StringType(thisStr,i)
+      TYPE(StringType),INTENT(INOUT) :: thisStr
+      INTEGER(SLK),INTENT(IN) :: i
+      CHARACTER(LEN=20) :: l
+      WRITE(l,'(i20)') i
```

Comment:
Similar question here about the number of digits for the 64-bit integer.

Also do not forget leaving an extra place for negative numbers.

Reply:
I checked and the widths do include space for the "-" sign.